themes/qgrid/source/js/menu.js

```diff
@@ -15,46 +17,233 @@ function toggleVisibility(e) {
 	}
 }
 
-function filter(e) {
+function onSearchChange(e) {
 	e.stopImmediatePropagation();
 	const input = e.target || e.srcElement;
-	const filter = input.value.toLowerCase();
+	const search = input.value.toLowerCase();
+	setSearch(search);
+}
+
+function searchOnLoad() {
+	const active = document.querySelector('.active-topic');
+	const search = getSearch();
+	if (active && search) {
+		document.getElementById('search').value = search;
+		updateMenuLinks(search);
+		currentPageSearch(search);
+		menuItemsSearch(search);
+	}
+}
+
+function searchOnHashChange() {
+	const search = getSearch();
+	document.getElementById('search').value = search;
+	updateMenuLinks(search);
```

Comment:
add a new line, new lines in function used:
1. to make it pretty code
2. to separate group of logic